File: src/main/java/model/rules/blokedgame/SmallestHandOpponentsPointSingle.java
package model.rules.blokedgame;

import model.players.Player;
import model.RoundResult;
import java.util.ArrayList;
import java.util.List;

public class SmallestHandOpponentsPointSingle implements BlockedGame {

    @Override
    public RoundResult execute(List<Player> players) {
        List<Integer> sums = new ArrayList<>();
        players.stream().map(p -> p.getTiles()).forEach(tiles -> sums.add(tiles.stream().mapToInt(t -> t.sum()).sum()));
        int min = sums.stream().min(Integer::compare).get();
        long count = sums.stream().filter(p -> p == min).count();
        if (count > 1) {
            return new RoundResult(RoundResult.DRAW, 0);
        }
        int minIndex = sums.indexOf(min);
        int opponentsSum = sums.stream().reduce(0, (a, b) -> a + b) - min;
        return new RoundResult(minIndex, opponentsSum);
    }
}

File: docs/scripts/model/rules/blockedgame/SmallestHandOpponentsPointSingle.js
import Result from "../../Result.js";
import ResultType from "../../ResultType.js";

export default class SmallestHandOpponentsPointSingle {

    execute(players) {
        let sums = [];
        players.map(p => p.getTiles()).forEach(tiles => sums.push(tiles.map(t => t.sum()).reduce((a, b) => a + b, 0)));
        let min = Math.min(...sums);
        let count = sums.filter(p => p === min).length;
        if (count > 1) {
            return new Result(ResultType.DRAW);
        }
        let minIndex = sums.indexOf(min);
        let opponentsSum = sums.reduce((a, b) => a + b, 0) - min;
        return new Result(minIndex, opponentsSum);
    }
}
